packages/web3/src/signer/signer.ts

```diff
@@ -318,6 +288,22 @@ export abstract class SignerWithNodeProvider implements SignerProvider {
   abstract signRaw(signerAddress: string, hexString: string): Promise<string>
 }
 
+export abstract class SignerProviderWithMultipleAccounts extends SignerProvider {
```

Comment:
Is the only non multiple accounts `SingerProvider` the `PrivateKeyWallet`? If that is the case would it make sense to have `SignerProvider` support multiple accounts by default and throw exception when `PrivateKeyWallet` wants to set the selected address? Since the user of the lib would probably always use this one instead. WDYT?

Reply:
The methods in this class are less popular: `getAccounts, setSelectedAccount`. WC wallet and extension wallet would not support them, so I think it's good to leave them outside `SignerProvider`. WDTY?

Reply:
Good point, then I agree with this approach. Thanks!